src/smartactions/SchedulingBase.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+import { IERC20 } from "forge-std/interfaces/IERC20.sol";
+import { IERC7579Execution } from "modulekit/Accounts.sol";
+import { ERC7579ExecutorBase } from "modulekit/Modules.sol";
+
+abstract contract SchedulingBase is ERC7579ExecutorBase {
+    error InvalidExecution();
+
+    event ExecutionAdded(address indexed smartAccount, bytes32 indexed executionHash);
+
+    event ExecutionTriggered(address indexed smartAccount, bytes32 indexed executionHash);
+
+    event ExecutionCancelled(address indexed smartAccount, bytes32 indexed executionHash);
+
+    mapping(address smartAccount => mapping(bytes32 jobHash => ExecutionConfig)) internal
+        executionLog;
+
+    struct ExecutionConfig {
+        uint48 executeInterval;
+        uint16 numberOfExecutions;
+        uint16 numberOfExecutionsCompleted;
+        uint48 startDate;
+        bool isCancelled;
+        uint48 lastExecutionTime;
+        bytes executionData;
+    }
+
+    modifier canExecute(bytes32 jobHash) {
```

Comment:
modifier should call internal function that does that check. will save deployment size / cost